app/tests/reader_studies_tests/test_views.py

```diff
@@ -696,6 +696,22 @@ def test_display_set_interfaces_create_permissions(client):
     )
     assert response.status_code == 200
 
+    response = get_view_for_user(
+        viewname="reader-studies:display-set-new-interfaces-create",
+        client=client,
+        reverse_kwargs={"slug": rs.slug},
+        user=u1,
+    )
+    assert response.status_code == 200
+
+    response = get_view_for_user(
+        viewname="reader-studies:display-set-new-interfaces-create",
+        client=client,
+        reverse_kwargs={"slug": rs.slug},
+        user=u2,
+    )
+    assert response.status_code == 200
```

Comment:
Why is a normal user getting a 200 response here? They shouldn't have any need to access this view, right?

Reply:
That's because the permission object is None here, then everyone passes the permission check for the view. 

Maybe the way to go would be to check change permissions for the reader study rather than the display set. The display set might not exist, but the reader study will always exist when this view is called, so we wouldn't end up with the permission object being None. Does that sound good? Or, we could make both the permission and the permission object dependent on whether there is an object and check reader study change permission when there is not and display set change permission when there is.

Reply:
> That's because the permission object is None here, then everyone passes the permission check for the view

Then, if I understand this correctly, that's a serious bug. Anyone could then add display sets to any reader study. This pattern needs to change everywhere where it occurs.

If there is no existing display set then the user needs to have permission to add display sets for that reader study, which may be being done with the change reader study object permission.

If there is an existing display set then the user needs to have permission to add a value to that display set, which may be being done with the change display set object permission.

Reply:
We're talking about the InterfaceCreate view here, which is a TemplateView, no adding or updating is done through that view. We do check permissions correctly on the create and update views. 

I will make the changes to the permission handling for the interface view as well though.

Reply:
Permission checks on the interface views should now be complete. I have also added a more comprehensive test.

Reply:
Thanks!